pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -99,6 +103,14 @@ func (e *HookExecutor) executeExecNewPod(hook *deployapi.LifecycleHook, deployme
 		return err
 	}
 
+	config, err := deployutil.DecodeDeploymentConfig(deployment, e.Codec)
```

Comment:
I thought of another simplification. Move the merge logic moved into `makeHookPod`. You could introduce a `DeploymentStrategy` argument like:

```go
func makeHookPod(hook *deployapi.LifecycleHook, strategy *deployapi.DeploymentStrategy, deployment *kapi.ReplicationController, label string) (*kapi.Pod, error) {
```

Decode the config in `executeExecNewPod` and pass on the strategy to `makeHookPod`.

Then you can very easily just update the fixture in `lifecycle_test.go` (via `deployment()`) and modify an existing scenario in `TestHookExecutor_makeHookPod` to add in your expected label/annotation maps. The object diffing in there will take care of all the assertions.

You can then delete `TestHookExecutor_useStrategyLabels`.